src/currency-field/CurrencyField.ts

```diff
@@ -0,0 +1,377 @@
+
+            let centsPart = this._parseCents(inputValue.substring(inputValue.indexOf(this._currencyCentsSeparator) + 1));
+
+            if (centsPart.length >= 2) {
+                centsPart = centsPart.substring(0, 2);
+            }
+            // Format amount and cents to currency string, ignoring cents if still partially completed eg: just '.' is valid.
+            this._stringValue = this._formatToCurrency(amountPart) + this._currencyCentsSeparator + centsPart;
+        } else {
+            this._stringValue = this._formatToCurrency(this._parseAmount(inputValue));
+        }
+
+        this.requestUpdate();
+
+        if (this._stringValue.includes(this._currencyFormatSeparator)) {
+            valueFormatterCount = this._stringValue.match(new RegExp(this._currencyFormatSeparator, 'g')).length;
+        }
+        await this.updateComplete;
+
+        // Set caret position after value is formatted into currency
+        if (valueLength < this._stringValue.length) {
+            const difference = this._stringValue.length - valueLength;
+            this._inputElement.setSelectionRange(caretPosition + difference, caretPosition + difference);
+        } else {
+            this._inputElement.setSelectionRange(caretPosition, caretPosition);
+        }
+
+        if (valueFormatterCount < formatterCount) {
+            this._inputElement.setSelectionRange(caretPosition - 1, caretPosition - 1);
+        }
+
+        // Set value prop to float value
+        this.value = this._formatToFloat(this._stringValue);
+    }
+
+    static override get styles() {
+        return [
+            super.styles,
+            css`
+                .field {
+                    flex: 1 1 auto;
+
+                    border: none;
+                    background: none;
+                    box-shadow: none;
+                    outline: 0;
+                    margin: 0;
+
+                    text-align: var(--omni-currency-field-text-align, left);
+
+                    color: var(--omni-currency-field-font-color, var(--omni-font-color));
+                    font-family: var(--omni-currency-field-font-family, var(--omni-font-family));
+                    font-size: var(--omni-currency-field-font-size, var(--omni-font-size));
+                    font-weight: var(--omni-currency-field-font-weight, var(--omni-font-weight));
+                    height: var(--omni-currency-field-height, 100%);
+                    padding: var(--omni-currency-field-padding, 10px);
+                }
+
+                .currency-symbol {
+                    font-size: var(--omni-currency-field-symbol-font-size, var(--omni-font-size));
+                    color: var(--omni-currency-field-symbol-color, var(--omni-font-color));
+                    padding-left: var(--omni-currency-field-symbol-left-padding, 10px);
+                    user-select: text;
```

Comment:
```suggestion
                    user-select: var(--omni-currency-field-symbol-select, text);
```

Consider making a variable?